sybilSig.go

```diff
@@ -0,0 +1,269 @@
+		IssuerClaimNonRevState:      s.IssuerClaim.NonRevProof.TreeState.State,
+
+		IssuerClaimSchema: s.IssuerClaim.Claim.GetSchemaHash().BigInt().String(),
+
+		StateCommitmentClaim:           s.StateCommitmentClaim.Claim,
+		StateCommitmentClaimMtp:        CircomSiblings(s.StateCommitmentClaim.IncProof.Proof, s.GetMTLevel()),
+		StateCommitmentClaimClaimsRoot: s.StateCommitmentClaim.IncProof.TreeState.ClaimsRoot,
+		StateCommitmentClaimRevRoot:    s.StateCommitmentClaim.IncProof.TreeState.ClaimsRoot,
+		StateCommitmentClaimRootsRoot:  s.StateCommitmentClaim.IncProof.TreeState.ClaimsRoot,
+		StateCommitmentClaimIdenState:  s.StateCommitmentClaim.IncProof.TreeState.ClaimsRoot,
+
+		GistRoot: s.GISTProof.Root,
+		GistMtp:  CircomSiblings(s.GISTProof.Proof, s.GetMTLevel()),
+
+		CRS: s.CRS,
+
+		UserGenesisID:            s.ID.BigInt().String(),
+		ProfileNonce:             s.ProfileNonce.String(),
+		ClaimSubjectProfileNonce: s.ClaimSubjectProfileNonce.String(),
+	}
+	uniAux := GetNodeAuxValue(s.IssuerClaim.NonRevProof.Proof)
+	sigInputs.IssuerClaimNonRevMtpNoAux = uniAux.noAux
+	sigInputs.IssuerClaimNonRevMtpAuxHi = uniAux.key
+	sigInputs.IssuerClaimNonRevMtpAuxHv = uniAux.value
+
+	gistAux := GetNodeAuxValue(s.GISTProof.Proof)
+	sigInputs.GistMtpAuxHi = gistAux.key
+	sigInputs.GistMtpAuxHv = gistAux.value
+	sigInputs.GistMtpNoAux = gistAux.noAux
+
+	issuerAuthAux := GetNodeAuxValue(s.IssuerClaim.SignatureProof.IssuerAuthNonRevProof.Proof)
+	sigInputs.IssuerAuthClaimNonRevMtpNoAux = issuerAuthAux.noAux
+	sigInputs.IssuerAuthClaimNonRevMtpAuxHi = issuerAuthAux.key
+	sigInputs.IssuerAuthClaimNonRevMtpAuxHv = issuerAuthAux.value
+
+	sigInputs.RequestID = s.RequestID
+	sigInputs.IssuerID = s.IssuerClaim.IssuerID.BigInt().String()
+	sigInputs.Timestamp = s.Timestamp
+
+	return json.Marshal(sigInputs)
+}
+
+type SybilSigPubSignals struct {
+	BaseConfig
+
+	SybilID *big.Int `json:"sybilID"`
+	UserID  *core.ID `json:"userID"`
+
+	RequestID *big.Int `json:"requestID"`
+	IssuerID  *core.ID `json:"issuerID"`
+	Timestamp int64    `json:"timestamp"`
+
+	IssuerClaimNonRevState *merkletree.Hash `json:"issuerClaimNonRevState"`
+
+	IssuerClaimSchema core.SchemaHash `json:"issuerClaimSchema"`
+
+	CRS *big.Int `json:"crs"`
+
+	GISTRoot *merkletree.Hash `json:"gistRoot"`
+
+	IssuerAuthState *merkletree.Hash `json:"issuerAuthState"`
+}
+
+func (s *SybilSigPubSignals) PubSignalsUnmarshal(data []byte) error {
+	var sVals []string
+	err := json.Unmarshal(data, &sVals)
+	if err != nil {
+		return err
+	}
+
+	if len(sVals) != 10 {
+		return fmt.Errorf("invalid number of Output values expected {%d} got {%d} ", 10, len(sVals))
+	}
+
+	// expected order:
+	//	0 - userID,
+	//	1 - sybilID,
+	//	2 - issuerAuthState,
+	//	3 - requestID,
+	//	4 - issuerID,
+	//	5 - timestamp,
+	//	6 - issuerClaimNonRevState,
+	//  7 - issuerClaimSchema
+	//	8 - crs,
+	//	9 - gistRoot,
+
+	if s.UserID, err = idFromIntStr(sVals[0]); err != nil {
```

Comment:
instead of hardcoded indexes I would use approach with `fieldIdx++`, will be easier to modify inputs if we need to add or remove some signals

Reply:
done